libp2p/services/autorelayservice.nim

```diff
@@ -115,19 +122,23 @@ proc innerRun(self: AutoRelayService, switch: Switch) {.async.} =
         break
       self.relayPeers[relayPid] = self.reserveAndUpdate(relayPid, switch)
 
-    if self.relayPeers.len() > 0:
+    try:
```

Comment:
I'm not sure we should replace this condition by a try/except, but instead add to it with something like

```nim
if self.relayPeers.len() > 0:
   try:
       await one(toSeq(self.relayPeers.values())) or self.peerAvailable.wait()
   except ValueError:
       raiseAssert "checked with relayPeers.len()" 
else:
   await self.peerAvailable.wait()
```

This is because raising an exception is expensive compared to checking the condition and acting based on the condition (since `relayPeers.len() is != 0`, then `one(` will not raise the exception).